docs/developers/enabling-ha.md

```diff
@@ -0,0 +1,75 @@
+  - [Configuring the leader election](#configuring-the-leader-election)
+- [Disabling leader election](#disabling-leader-election)
+  - [Use the disable-ha flag](#use-the-disable-ha-flag)
+  - [Scale down your replicas](#scale-down-your-replicas)
+
+## Overview
+
+---
+This document is aimed at helping maintainers/developers when configuring High Availability(HA) support for the Tekton Pipeline [controller deployment](../../config/controller.yaml).
+
+HA support allows components to remain operational when a disruption occurs. This is achieved by following an active/passive model, where only one instance is active at a time.
+
+In the case of the Tekton pipeline controller, when the active controller fails or is deleted, another instance of that controller should become the active one, taking over all related API requests that the previous active controller was serving. The active instance is usually refer as the `leader`.
+
+By default HA is enable in the Tekton pipelines controller.
+
+## Configuring HA
+
+---
+In order to achieve HA, the number of replicas for the Tekton Pipeline controller should be greater than one. This allows other instance(_s_) to take over in case of any disruption on the current active controller.
+
+### Configuring the controller replicas
+
+You can modify the replicas number in the [controller deployment](../../config/controller.yaml) under `spec.replicas` or apply an update to a running deployment:
+
+```sh
+kubectl -n tekton-pipelines scale deployment tekton-pipelines-controller --replicas=3
+```
+
+### Configuring the Leader Election
+
+The leader election can be configured via the [config-leader-election.yaml](../../config/config-leader-election.yaml). The configmap defines the following parameters:
+
+| Parameter            | Default  |
+| -------------------- | -------- |
+| `data.resourceLock`  | "leases" |
+| `data.leaseDuration` | 15s      |
+| `data.renewDeadline` | 10s      |
+| `data.retryPeriod`   | 2s       |
+
+
+## Disabling leader election
+
+---
+
+If HA is not required and running a single instance of the Tekton Pipeline controller is enough, there are two alternatives:
+
+### Use the disable-ha flag
+
+You can modify the [controller deployment](../../config/controller.yaml), by specifying in the `tekton-pipelines-controller` container the `disable-ha` flag. For example:
+
+```yaml
+spec:
+  serviceAccountName: tekton-pipelines-controller
+  containers:
+  - name: tekton-pipelines-controller
+    ...
+    args: [
+      # Other flags defined here...
+      "-disable-ha=true"
+    ]
+```
+
+by setting `disable-ha` to `true`, HA will be disable in the controllers.
+
+**Note**: Please consider that when disabling HA and keeping multiple replicas of the [controller deployment](../../config/controller.yaml), each replica will act as an independent controller, leading to an unwanted behaviour when creating resources ( e.g. `TaskRuns`, etc )
```

Comment:
Nice! Thanks for making a note about this!

NIT: missing dot at the end of the line, extra space before after `(` and before `)`

Reply:
fixed